Userland/Libraries/LibDSP/Resampler.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2023, Sarsaparilla
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Array.h>
+#include <AK/Math.h>
+#include <AK/OwnPtr.h>
+#include <AK/Span.h>
+#include <AK/Vector.h>
+#include <LibDSP/FIRFilter.h>
+#include <LibDSP/Window.h>
+
+namespace DSP {
+
+class PreciseSinc {
+public:
+    PreciseSinc(size_t = 0, size_t = 0)
+    {
+    }
+
+    double operator()(double phi) const
+    {
+        if (AK::abs(phi) < 0.00000001) {
+            return 1.0;
+        } else {
+            phi *= AK::Pi<double>;
+            return AK::sin(phi) / phi;
+        }
+    }
+};
+
+class InterpolatedSinc {
+public:
+    InterpolatedSinc(size_t taps, size_t oversample)
```

Comment:
Just to make sure: Meaning create a static `make` method which returns a `ErrorOr<InterpolatedSinc>`?
Cause otherwise this won't work because of the vector.

Reply:
Yep, the vector is the problem here. Again we'd want a constexpr-able version that can use Array instead of Vector (since then the sizes are known at compile time).